Ambari 9016 branch 2.7

payert opened the pull request:
## What changes were proposed in this pull request?
Ambari almost always responds with "Content-Type:text/plain" even if the response content itself is a JSON. Also, if Ambari is requested with "Content-Type:application/json" if fails with an exception.
This changes proposes a filter that:
-  Change the Content-Type of a **request** to **text/plain** if the **original Content-Type** is **application/json** to prevent Ambari failing on the application/json content.

- Change to response's Content-Type to application/json if the **original request's Content-Type** is **application/json**

## How was this patch tested?
- Unit test has been added.
- Unit tests of ambari-server have been run along with manual testing.
- Parts HDP certification was run.

Commit:
BUG-105818 Ambari API uses HTTP Header Content-Type:text/plain when the content is JSON.

Ambari should not fail if the request is using 'Content-Type:application/json'
A filter overrides the 'application/json' content type to text/plain.

Change-Id: Ieb38e012c1a82f742343ecb8a539f73df457ba96

Commit:
BUG-105818 Ambari API uses HTTP Header Content-Type:text/plain when the content is JSON.

If the Content-Type of the request is "application/json", the Content-Type of response
is changed to "application/json" from "text/plain".

Change-Id: Ic4bf567af4ce20132c4e66fb7e2dbcbab0806292

Commit:
BUG-105818 Ambari API uses HTTP Header Content-Type:text/plain when the content is JSON.

Javadoc updated.

Change-Id: Icbda33e3e588c30b9bd4ab44b78bd2775db5e1b1

Commit:
AMBARI-9016 BUG-105818 Discovering the excluded URLs added

Change-Id: If618d57735fecf3381a50d9bc64d33610cf3260c

Commit:
AMBARI-9016 BUG-105818 Fix checksyle error

Change-Id: I15f144b6805bf6488be090424d0405b4b35874e0

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5299/
Test FAILed.
Test FAILured.

kasakrisz reviewed (approved):
LGTM

Commit:
AMBARI-9016 BUG-105818 Add license header

Change-Id: I98208f2be51aca904f8a33b00e2970eae53f4a98

g-boros commented:
@dlysnichenko  @mpapirkovskyy  @aonishuk can any of you pls review?

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5304/
Test FAILed.
Test FAILured.

payert commented:
retest this please

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5306/
Test PASSed.

Commit:
AMBARI-9016 BUG-105818 Fixed java package location and changed inner class visibility.
- ContentTypeOverrideFilter was moved to java.org.apache.ambari.server.api package.
- Inner Response/Request wrapper classes visibility has changed to package-private
  to support unit testing.

Change-Id: I41e8aff8ac8c1782e812f84aa9e71c5e055521df

Commit:
AMBARI-9016 BUG-105818 Unit test added

Change-Id: I89409ee22c3a1c44b51772407f851fb05d987bb0

Commit:
AMBARI-9016 BUG-105818 Fix checkstyle errors

Change-Id: Ib6c7d1430c67f1d839f19850d3736a366d2b481b

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/5320/
Test PASSed.
